File: Calculator.py
def add(a, b):
    return a + b

def sub(a, b):
    return a - b

def multiply(a, b):
    return a * b

def divide(a, b):
    if b == 0:
        return "ERROR, b can't be zero!"
    return a / b

while True:
    try:
        a = float(input("Enter the first number: "))
        b = float(input("Enter the second number: "))
        sign = input("Enter the operation (+, -, *, /) or type 'exit' to quit: ")

        if sign == "exit":
            print("Exiting the calculator. Bye!")
            break

        if sign == "+":
            print("Result:", add(a, b))
        elif sign == "-":
            print("Result:", sub(a, b))
        elif sign == "*":
            print("Result:", multiply(a, b))
        elif sign == "/":
            print("Result:", divide(a, b))
        else:
            print("Invalid operator. Please use +, -, *, / or 'exit'.")
    except ValueError:
        print("Invalid input! Please enter numbers only.")

File: Calculator.java
import java.util.Scanner;
class Calculator {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int ans =0;
        while(true){
        System.out.print("enter the operator: ");
        char op = input.next().trim().charAt(0);
        if (op=='+'|| op=='-'|| op=='*'||op=='/'|| op =='%'){
            System.out.println("enter the 2 numbers: ");
            int a = input.nextInt();
            int b = input.nextInt();
            if (op=='+'){
                ans = a+b;
            }
            if  (op=='-'){
                ans = a-b;
            }
            if  (op=='*'){
                ans = a*b;
            }
            if  (op=='/'){
                if (b!=0){
                    ans = a/b;
                }
            }
            if  (op=='*'){
                ans = a*b;
            }
        }else if (op=='x'||op=='X'){
            break;
        }else{
            System.out.println("invalid operator");
        }
        System.out.println("therefore the answer is " + ans);
    }
}
    
}